documentation/projects/proposals/additional_search_views/20230719-implementation_plan_additional_search_views.md

```diff
@@ -111,35 +118,156 @@ for fuzzy matches.
+#### Search request serializer updates
+
+_The initial version of this plan proposed to add new API endpoints and use path
+parameters for the values (e.g., `/image/tag/cat`) to make the URLs more
+readable, easier to share, will be easier to cache or perform cache invalidation
+required by [#1969](https://github.com/WordPress/openverse/issues/1969).
+However, since tags and creator names contain characters that are special for
+path segments (`/`, `?` and `&`) that cannot be properly encoded, this approach
+turned out to be not feasible._
+
+```{note}
+The new query parameters (`collection` and `tag`) will have `unstable__` prefix until this project is
+launched. In the text below, for brevity, the prefix is omitted.
+```
+
+The collections will use the `search` endpoint with `collection` query parameter
+set to `tag`, `creator` or `source`, and the relevant query parameters. The
+`collection` parameter validator will check that the request contains the
+necessary additional parameters: `source` for `collection=source`, `creator` and
+`source` for `collection=creator`, and `tag` for `collection=tag`. The value of
+the `source` parameter will also be checked to be an existing source using the
+provider store.
+
+For the tag route, the new singular `tag` parameter should be used, rather than
+the existing plural `tags` should be used for legibility since we are presenting
+a single tag.
+
+The existing `source` and `creator` parameters will be reused, but will be
+parsed differently when `collection` parameter is present: they will only allow
+a single value instead of being split by `,` as it is for the default search.
+
+Their documentation should be updated to reflect that. The additional
+documentation for the search parameters should be added as a draft so that it's
+not published on the API documentation site until we launch this project and
+remove the `unstable__` prefix.
+
+### 2. Add the `additional_search_views` feature flag
+
+The flag should be switchable, and off by default.
+
+### 3. Nuxt store and API request changes
+
+We can reuse the search store as is for these pages.
+
+_Previously, frontend search store had a `searchBy` filter that allowed to
+search within the `creator` field. When `searchBy` value was set, the API `q`
+parameter was replaced with the `<searchBy>=<searchTerm>` API query parameter.
+This filter was removed because `searchBy` is not strictly a filter that can be
+toggled on or off_
+
+#### Add `strategy` and `collectionParams` to `search` store
+
+The `strategy` parameter will be used to determine the API request query
+parameters. If it is set to `"search"`, then the API query will be created using
+the current approach. If it is set to `"collection"`, the query will be
+constructed using the new method (`buildCollectionQuery`) to set `collection`
+and other relevant parameters using the new `collectionParams` object in the
+store. It will _not_ be setting the filter parameters such as license or
+category.
```

Comment:
What happens if unsupported params are present in the URL, are they ignored or do they they throw HTTP errors?

Reply:
Added that they are ignored.